### PR TITLE
WRFDA for racm_soa_vbs_da (chem_opt=108) in the WRF-Chem model

### PR DESCRIPTION
**Pull request by syha**
TYPE: new features

KEYWORDS: aerosol data assimilation, RACM, MADE-VBS aerosols, WRF-Chem

SOURCE: Soyoung Ha (MMM/NCAR)

DESCRIPTION OF CHANGES: 
WRFDA is newly expanded for chem_cv_options=108 for surface chemical data assimilation (PM2.5, PM10, CO, SO2, NO2, O3) using the RACM/MADE-VBS chemistry option in WRF-Chem.

LIST OF MODIFIED FILES: 
M       Registry/registry.var_chem
M       var/da/da_chem_sfc/da_ao_stats_chem_sfc.inc
M       var/da/da_chem_sfc/da_calculate_grady_chem_sfc.inc
M       var/da/da_chem_sfc/da_chem_sfc.f90
M       var/da/da_chem_sfc/da_get_innov_vector_chem_sfc.inc
M       var/da/da_chem_sfc/da_residual_chem_sfc.inc
M       var/da/da_chem_sfc/da_transform_xtoy_chem_sfc.inc
M       var/da/da_chem_sfc/da_transform_xtoy_chem_sfc_adj.inc
M       var/da/da_define_structures/da_allocate_observations_chem_sfc.inc
M       var/da/da_define_structures/da_deallocate_background_errors.inc
M       var/da/da_define_structures/da_define_structures.f90
M       var/da/da_main/da_update_firstguess.inc
M       var/da/da_obs_io/da_final_write_obs_gas_sfc.inc
M       var/da/da_obs_io/da_read_obs_chem_sfc.inc
M       var/da/da_obs_io/da_read_omb_tmp.inc
M       var/da/da_obs_io/da_scan_obs_chem_sfc.inc
M       var/da/da_obs_io/da_write_obs_chem_sfc.inc
M       var/da/da_setup_structures/da_get_vertical_truncation.inc
M       var/da/da_setup_structures/da_setup_background_errors.inc
M       var/da/da_setup_structures/da_setup_be_regional.inc
M       var/da/da_setup_structures/da_setup_firstguess_wrf.inc
M       var/da/da_setup_structures/da_setup_obs_structures.inc
M       var/da/da_setup_structures/da_setup_obs_structures_chem_sfc.inc
M       var/da/da_statistics/da_analysis_stats.inc
M       var/da/da_statistics/da_print_qcstat.inc
M       var/da/da_transfer_model/da_transfer_model.f90
M       var/da/da_transfer_model/da_transfer_xatoanalysis.inc
M       var/da/da_vtox_transforms/da_transform_vchemtox_adj.inc

TESTS CONDUCTED: 
1. All tests were passed with no issues.
2. Jenkins testing is OK
3. WRFDA regression tests passed.

RELEASE NOTE: WRFDA is newly expanded for chem_cv_options=108 for surface chemical data assimilation (PM2.5, PM10, CO, SO2, NO2, O3) using the RACM/MADE-VBS chemistry option in WRF-Chem.
Ha, Soyoung, 2021: Implementation of aerosol data assimilation in WRFDA (V4.0.3) for WRF-Chem (V3.9.1) using the MADE/VBS scheme. Geoscientific Model Development. gmd-2021-74.

### CODE REVIEW

**Comment by davegill** [comment]
@syha @jordanschnell 
Soyoung,
HI!
1. Is this an entirely new feature? If so, could you put in a simple figure that shows - yes we are assimilating some field.
2. Would you add a DOI to the publication please.

Jordan,
Is there a README_of_all_chem_options.txt file that Soyoung should also update, even though this is DA?

**Comment by davegill** [comment]
@syha 
Soyoung,
Your list has 29 files modified, the github count is 28. Please verify the list that you provided.

**Comment by syha** [comment]
> @syha @jordanschnell Soyoung, HI!
> 
> 1. Is this an entirely new feature? If so, could you put in a simple figure that shows - yes we are assimilating some field.
> 2. Would you add a DOI to the publication please.
> 
> Jordan, Is there a README_of_all_chem_options.txt file that Soyoung should also update, even though this is DA?

Hi, Dave,

1. Yes, it is a brand-new feature introduced in WRFDA for WRF-Chem. I'm not sure what kind of a simple figure you meant, though. A schematic diagram or a sample result figure? I couldn't find one from Jake's similar implementation for V43.
2. The manuscript is in the final stage of the review process (since March 2021 - can you believe it? It takes forever), so hopefully I can get a DOI before the release (or freeze?).
3. I just added the file that had been missed in my yesterday's push. Also, I realized that there is no namelist sample or README.namelist for WRF-Chem/WRFDA anywhere in WRF, so I just created one (with my new namelist setting) in WRF/var/run/namelist.input. As you said, we may need to make a new sample namelist and doc such as namelist.input.var_chem and Readme.var_chem under WRF/run/. If we do that, my sample in WRF/var/run/namelist.input can be merged and I can contribute to the README doc, as well. Just let me know.

**Comment by davegill** [comment]
@syha
Soyoung,
1. I am looking for a plot that shows the impact of something being assimilated vs not. If this does not exist - no biggy.
2. If there is no DOI, there is no DOI - again no biggy. Perhaps mention my name to the journal, that might pick up the pace a bit. :)
3. We'll take a look at the namelist options that you are providing regarding some documentation.

Usually these DA PRs are handled pretty quickly, as opposed to other PRs that may hang around for weeks or months.

**Comment by liujake** [comment]
I will review this soon. Given the number of files modified, it will take some time. We have a chem-da chapter in WRFDA's user guide, but it will be good if we can add a readme file as well for chem-da options.

**Comment by davegill** [comment]
@liujake @syha @weiwangncar @dudhia @jordanschnell 
Jake,
No rush at all.

Jake and Soyoung,
We all like the idea of a simple DA-chem options README. There is a doc directory in WRF, and that might be a good place for it. 

Jordan, Wei, Jimy,
Opinions on a proposed README doc location? Do we want to have a separate PR for that first?

File: var/da/da_chem_sfc/da_transform_xtoy_chem_sfc_adj.inc
Comment (liujake) [inline var/da/da_chem_sfc/da_transform_xtoy_chem_sfc_adj.inc:25]: Same comment as for tangent linear code. Can remove this line.

File: var/run/namelist.input
Comment (liujake) [inline var/run/namelist.input:5]: The namelist.input file should be removed. I will add a readme file for chem_da in another PR.

**Review by liujake — CHANGES_REQUESTED** [review]
Overall looks good! I will make some regression tests after my comments are addressed.

File: Registry/registry.var
Comment (syha) [reply Registry/registry.var:303]: Sounds good to me.

File: var/da/da_chem_sfc/da_transform_xtoy_chem_sfc.inc
Comment (syha) [reply var/da/da_chem_sfc/da_transform_xtoy_chem_sfc.inc:23]: Agree.

File: var/da/da_chem_sfc/da_transform_xtoy_chem_sfc_adj.inc
Comment (syha) [reply var/da/da_chem_sfc/da_transform_xtoy_chem_sfc_adj.inc:25]: Agree

**Comment by liujake** [comment]
@syha You may commit changes on your branch.

**Comment by liujake** [comment]
@syha @davegill I tried to build/test this branch, but I had trouble to compile this branch's code for the WRF model (./compile em_real) with the error message below:
```
wrf_shutdown.f90:4.21:

   USE , INTRINSIC :: IEEE_EXCEPTIONS
                     1
Fatal Error: Can't find an intrinsic module named 'ieee_exceptions' at (1)
0.04user 0.11system 0:00.11elapsed 131%CPU (0avgtext+0avgdata 6264maxresident)k
0inputs+0outputs (0major+12885minor)pagefaults 0swaps
../configure.wrf:351: recipe for target 'wrf_shutdown.o' failed
make[2]: [wrf_shutdown.o] Error 1 (ignored)
```

'develop' branch code has no problem to compile WRF model. Modified code are all for DA, not sure why this happens. Any ideas?
WRF model build is the first step of the whole WRFDA regression test procedures.

**Comment by liujake** [comment]
Failed build is the compilation from a PBS job. When I rebuilt the code directly from terminal, it built Ok. Must be caused by some changes in Cheyenne environment. It was Ok previously to build from a PBS job.

**Review by liujake — APPROVED** [review]

**Comment by liujake** [comment]
@davegill This PR is ready to merge, but looks like I have no permission to merge.

**Review by kkeene44 — APPROVED** [review]

**Comment by kkeene44** [comment]
@liujake 
I looked at permissions and don't see any reason why you wouldn't be able to merge it, except that we changed our rules so that we now have to have two approvals for each PR. I just approved it, so let me know if you have access to merge it now.

**Comment by liujake** [comment]
Screen Shot 2021-11-19 at 10 14 16 AM
This is what I see now even with your approval.

**Comment by kkeene44** [comment]
@liujake 
Okay, I think I know what the problem was. The last time we released, the incorrect group was assigned to merging rights. I've corrected it now. Let me know if you see any difference.

**Comment by liujake** [comment]
Ok now. I am merge it now.

**Pull request merged**
